app/searchers/responses_searcher.rb

```diff
@@ -162,16 +169,81 @@ def maybe_filter_by_expression(expression, op_kind, token_values, is_filterable)
+      return filter_by_questions(expression.qualifier_text, token_values)
+    when "reviewed"
+      return filter_by_boolean(token_values, filter_name: :is_reviewed)
+    when "submitter"
+      return filter_by_names(token_values, User, current_ids: submitters, include_name: true)
+    when "group"
+      return filter_by_names(token_values, UserGroup, current_ids: groups, include_name: true)
+    else
+      return false
     end
+  end
+
+  # Given a list of values, find all instances of this class that match,
+  # and append their IDs to the existing list of IDs to filter by.
+  # Returns false if unable to handle this case.
+  def filter_by_names(names, klass, current_ids: [], include_name: false)
+    matches = klass.where("name ILIKE ANY (array[?])", names).pluck(:id, :name)
+    return false if matches.empty?
+    current_ids.concat(matches.map { |id, name| include_name ? {id: id, name: name} : id })
+    true
+  end
+
+  # Filter by a valid boolean.
+  # Returns false if unable to handle this case.
+  def filter_by_boolean(token_values, filter_name:)
+    return false unless token_values.length == 1
+    value = token_values[0].downcase
+    return false unless %w[1 0 yes no].include?(value)
+    is_truthy = %w[1 yes].include?(value)
+    instance_variable_set("@#{filter_name}", is_truthy)
+    true
+  end
+
+  # Filter by a question code + question value.
+  # Returns false if unable to handle this case.
+  def filter_by_questions(qualifier_text, token_values)
+    return false unless token_values.length == 1
+    # Strip the surrounding {QuestionCode} braces.
+    question_code = qualifier_text[1..-2]
+    matched_question = Question.where("code ILIKE ?", question_code).first
+    return false if matched_question.blank?
+    matched_qings = Questioning.where(question: matched_question)
+    return false if matched_qings.blank?
+    # Get the deterministic Questioning ID from the Question
+    matched_qing_id = FilterDataController.filter_unique(matched_qings).first.id
+    value = qing_value(matched_question, token_values)
+    qings.concat([{id: matched_qing_id}.merge(value)])
+    true
+  end
+
+  # Get the qing value from the user input (either a string to match or an Option ID).
+  def qing_value(matched_question, token_values)
+    value = token_values[0]
+    return {value: value} unless matched_question.option_set_id
+    {option_node_id: find_option_node_id(matched_question.option_set_id, value)}
+  end
+
+  def find_option_node_id(option_set_id, value)
+    partial_match = OptionNode.joins(:option).includes(:option)
+      .where(option_set_id: option_set_id)
+
+    # TODO: Can this be more efficient?
+    # Doesn't work with jsonb: `partial_match.where("options.name_translations ILIKE ?", value)`
+    full_match = partial_match.where("options.canonical_name ILIKE ?", value) +
+      partial_match.select do |node|
+        node.option.name_translations.values.map(&:downcase).include?(value.downcase)
+      end
```

Comment:
^ Yuck, please let me know if you have a better way to do this. I couldn't find any way to match jsonb values, even though it seems like it should be trivial to cast them to a string and use `ILIKE`.

Reply:
I think you can use `JSONB_EACH_TEXT`. See `app/models/results/answer_search_vector_updater.rb`. 

I assume you're using ILIKE just to do a case insensitive comparison and are not using the wildcards (`%`) at all? If so, using `LOWER()` with regular comparison will be faster I think. `ILIKE` should only be used if you need wildcards in my experience, which is not true here.

Let me know if you want to pair on this one. If you can get something working fairly quickly go for it, else let's pair.

Reply:
Good call on lower vs. ilike, I'll update. I tried `JSONB_EACH_TEXT` previously but it fails with `set-returning functions are not allowed in WHERE`:

```ruby
    full_match = partial_match.where("options.canonical_name ILIKE ?", value)
      .or(partial_match.where("JSONB_EACH_TEXT(options.name_translations) ILIKE ?", value))
```

```console
ActiveRecord::StatementInvalid (PG::FeatureNotSupported: ERROR:  set-returning functions are not allowed in WHERE
LINE 1: ... $1 AND ((options.canonical_name ILIKE 'Cat') OR (JSONB_EACH...
                                                             ^
: SELECT "option_nodes"."id" FROM "option_nodes" INNER JOIN "options" ON "options"."id" = "option_nodes"."option_id" WHERE "option_nodes"."option_set_id" = $1 AND ((options.canonical_name ILIKE 'Cat') OR (JSONB_EACH_TEXT(options.name_translations) ILIKE 'Cat'))):
```

Reply:
We can pair this week if you think there's another way to get it working (maybe different syntax outside of `where`)